src/java.base/share/native/launcher/main.c

```diff
@@ -167,6 +167,49 @@ static void setup_sighandler() {
     }
 }
 
+static const char *last_pid_filename = "/proc/sys/kernel/ns_last_pid";
+
+static int get_last_pid() {
+    FILE *last_pid_file = fopen(last_pid_filename, "r");
+    if (!last_pid_file) {
+        perror("last_pid_file fopen");
+        return -1;
+    }
+    int last_pid;
+    if (0 >= fscanf(last_pid_file, "%d", &last_pid)) {
+        fclose(last_pid_file);
+        perror("last_pid_file fscanf");
+        last_pid = -1;
+    }
+    fclose(last_pid_file);
+    return last_pid;
+}
+
+static void set_last_pid(int pid) {
+    FILE *last_pid_file = fopen(last_pid_filename, "w");
+    if (!last_pid_file) {
+        perror("last_pid_file fopen");
+        return;
+    }
+    if (0 > fprintf(last_pid_file, "%d", pid)) {
+        perror("last_pid_file fprintf");
+    }
+    fclose(last_pid_file);
+}
+
+static void spin_last_pid(int pid) {
+    for (pid_t child = fork(); child < (pid_t)pid; child = fork()) {
+        if (0 == child) {
+            exit(0);
+        }
+        int status;
+        if (0 > waitpid(child, &status, 0)) {
+            perror("waitpid last pid");
+            break;
+        }
```

Comment:
Can we drop this completely and rely on wait_for_children() below in the control flow?

Reply:
Sure, however, will it be OK if we create hundreds or thousands of child processes at the same time, even if they exist soon?

Reply:
Performance-wise, it looks even better to collect them after java has started execution. But you're right, that will create a burst of short living processes. OK, probably this is a safer option.